src/main/java/org/jabref/gui/edit/automaticfiededitor/copyormovecontent/CopyOrMoveFieldContentTabViewModel.java

```diff
@@ -0,0 +1,182 @@
+package org.jabref.gui.edit.automaticfiededitor.copyormovecontent;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javafx.beans.binding.BooleanBinding;
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.property.SimpleObjectProperty;
+
+import org.jabref.gui.edit.automaticfiededitor.AbstractAutomaticFieldEditorTabViewModel;
+import org.jabref.gui.edit.automaticfiededitor.AutomaticFieldEditorEvent;
+import org.jabref.gui.edit.automaticfiededitor.MoveFieldValueAction;
+import org.jabref.gui.undo.NamedCompound;
+import org.jabref.gui.undo.UndoableFieldChange;
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.model.strings.StringUtil;
+
+import de.saxsys.mvvmfx.utils.validation.FunctionBasedValidator;
+import de.saxsys.mvvmfx.utils.validation.ValidationMessage;
+import de.saxsys.mvvmfx.utils.validation.ValidationStatus;
+import de.saxsys.mvvmfx.utils.validation.Validator;
+
+public class CopyOrMoveFieldContentTabViewModel extends AbstractAutomaticFieldEditorTabViewModel {
+    public static final int TAB_INDEX = 1;
+    private final ObjectProperty<Field> fromField = new SimpleObjectProperty<>(StandardField.ABSTRACT);
+
+    private final ObjectProperty<Field> toField = new SimpleObjectProperty<>(StandardField.AUTHOR);
+
+    private final BooleanProperty overwriteFieldContent = new SimpleBooleanProperty(Boolean.FALSE);
+    private final List<BibEntry> selectedEntries;
+    private final NamedCompound dialogEdits;
+
+    private final Validator toFieldValidator;
+
+    private final BooleanBinding canMove;
+
+    private final BooleanBinding canSwap;
+
+    public CopyOrMoveFieldContentTabViewModel(List<BibEntry> selectedEntries, BibDatabase bibDatabase, NamedCompound dialogEdits) {
+        super(bibDatabase);
+        this.selectedEntries = new ArrayList<>(selectedEntries);
+        this.dialogEdits = dialogEdits;
+
+        toFieldValidator = new FunctionBasedValidator<>(toField, field -> {
+            if (StringUtil.isBlank(field.getName())) {
+                return ValidationMessage.error("Field name cannot be empty");
+            } else if (StringUtil.containsAnyWhitespaceCharacters(field.getName())) {
```

Comment:
```suggestion
            } else if (field.getName().isBlank())) {
```

Reply:
But `String#isBlank` will return false for 'file url'

Reply:
Ah now I see, but then I would simply not use a regex but use a method which loops through the chars of the string and does Character.isWhitespace(char) test:

e.g.
You shoudl probably check with is
 ```
boolean containsWhiteSpace(String str) 
if(str != null )
for( int i; i<  str.toCharArray().length; i++)
if(Character.isWhitespace(str[i]) {
return true
}
return false
```